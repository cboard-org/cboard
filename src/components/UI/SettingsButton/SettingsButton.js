import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import SettingsIcon from '@material-ui/icons/Settings';

import IconButton from '../IconButton';
import messages from './SettingsButton.messages';

const propTypes = {
  /**
   * The component used for the root node. Either a string to use a DOM element or a component.
   */
  component: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.object
  ]),
  /**
   * If true, back button is disabled
   */
  disabled: PropTypes.bool,
  /**
   * @ignore
   */
  intl: intlShape.isRequired,
  /**
   * Callback fired when back button is clicked
   */
  onClick: PropTypes.func
};

function SettingsButton(props) {
  const { intl, ...other } = props;
  const label = intl.formatMessage(messages.settings);

  return (
    <IconButton label={label} {...other}>
      <SettingsIcon />
    </IconButton>
  );
}

SettingsButton.propTypes = propTypes;

export default injectIntl(SettingsButton);
