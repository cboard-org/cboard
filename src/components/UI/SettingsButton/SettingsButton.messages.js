import { defineMessages } from 'react-intl';

export default defineMessages({
  settings: {
    id: 'cboard.components.SettingsButton.settings',
    defaultMessage: 'Settings'
  }
});
