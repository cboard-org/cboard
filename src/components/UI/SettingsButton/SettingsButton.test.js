import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';

import SettingsButton from './SettingsButton';

describe('SettingsButton tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<SettingsButton onClick={() => {}} />);
  });
});
