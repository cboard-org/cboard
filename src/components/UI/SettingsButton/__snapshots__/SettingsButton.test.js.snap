// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsButton tests default renderer 1`] = `
<SettingsButton
  intl={
    Object {
      "formatMessage": [Function],
    }
  }
  onClick={[Function]}
/>
`;
